src/pmhn/_trees/_simulate.py

```diff
@@ -31,7 +36,38 @@ def _simulate_tree(
     # TODO(Pawel): This is part of https://github.com/cbg-ethz/pMHN/issues/14
     #   Note that the sampling time is known that our `theta` entries
     #   are log-Theta entries from the paper.
-    raise NotImplementedError
+    print("starting ... \n")
+    theta_size=len(theta)
+    node_time_map={}
+    root=Node("0")
+    node_time_map[root]=0
+    U_current=[root]
+    exit_while=False
+    while len(U_current)!=0:
+        U_next=[]
+        for node in U_current:
+            path=list(node.path)
+            old_mutations=[int(node.name) for node in path]
+            possible_mutations=list(set([i+1 for i in range(theta_size)]).difference(set(old_mutations)))
+            for j in possible_mutations:
+                new_node=Node(str(j),parent=node)
+                l=np.exp(theta[j-1][j-1])
+                for anc in [ancestor for ancestor in node.path if ancestor.parent is not None]:
+                    l*=np.exp(theta[j-1][int(anc.name)-1])
```

Comment:
Numerical stability hint: it's generally better to `np.exp` only in the end: 

$$\exp \sum_i t_i = \prod_i \exp t_i$$

Here I'm not sure whether it'll in the end make an observable difference, but it's better to be on the safe side here :slightly_smiling_face: